### ISSUE TITLE
feat: Cairo to JSON

### ISSUE THREAD

**Issue by caseywescott** [cienicera/Koji#31]
**A suite of functions mapping Cairo Midi msgs to JSON object for web retrieval.**

 Functions that format the following messages to JSON:

    NOTE_ON
    NOTE_OFF
    SET_TEMPO
    TIME_SIGNATURE
    CONTROL_CHANGE
    PITCH_WHEEL
    AFTER_TOUCH
    POLY_TOUCH

These functions can then be used create a function 'midi_to_json' that iterates through Midi msgs and formats them to create the complete JSON object.

**Comment by caseywescott** [comment]
I'm cobbling together a rough cut of cairo-to-json [Here](https://github.com/cienicera/Koji/blob/fa90bb93a17f265e55a4215d0fa56965cdaf659c/src/midi/core.cairo#L696). I have all of the necessary components for each message but need to serialize the values. Would a ByteArray be best?  I took a quick look at the ['format!' macro](https://book.cairo-lang.org/ch11-02-macros.html?highlight=format!#format-macro) and am hoping that could do the trick to simplify the code but I think it's unavailable in scarb/cairo 2.3.0. Have you used 'format!' for serializing?

**Comment by raphaelDkhn** [comment]
> I'm cobbling together a rough cut of cairo-to-json [Here](https://github.com/cienicera/Koji/blob/fa90bb93a17f265e55a4215d0fa56965cdaf659c/src/midi/core.cairo#L696). I have all of the necessary components for each message but need to serialize the values. Would a ByteArray be best? I took a quick look at the ['format!' macro](https://book.cairo-lang.org/ch11-02-macros.html?highlight=format!#format-macro) and am hoping that could do the trick to simplify the code but I think it's unavailable in scarb/cairo 2.3.0. Have you used 'format!' for serializing?

@caseywescott  Yes you're right I think format! macro would do the trick. I just opened an issue here to update to the last cairo version. https://github.com/cienicera/Koji/issues/35

**Comment by vibenedict** [comment]
Hi @caseywescott , I would like to try this out.

**Comment by caseywescott** [comment]
Go for it!!!

**Comment by caseywescott** [comment]
I'll assign to you when when i get home in 30mins

**Comment by caseywescott** [comment]
Just Assigned to you @vibenedict !!! Stoked! Let me know if you have any questions!!
